astropy/stats/funcs.py

```diff
@@ -790,3 +792,322 @@ def mad_std(data):
+    data = np.sort(data)
+    cdfv = cdf(data,*args)
+    N = len(data)
+    D = (np.amax(cdfv-np.arange(N)/float(N))
+         + np.amax((np.arange(N)+1)/float(N)-cdfv))
+
+    return D, kuiper_FPP(D,N)
+
+def kuiper_two(data1, data2):
+    """Compute the Kuiper statistic to compare two samples.
+
+    Parameters
+    ----------
+    data1 : array-like
+        The first set of data values.
+    data2 : array-like
+        The second set of data values.
+    
+    Returns
+    -------
+    D : float
+        The raw test statistic.
+    fpp : float
+        The probability of obtaining two samples this different from
+        the same distribution.
+
+    Notes
+    -----
+    Warning: the fpp is quite approximate, especially for small samples.
+
+    """
+    data1, data2 = np.sort(data1), np.sort(data2)
+
+    if len(data2)<len(data1):
+        data1, data2 = data2, data1
+
+    cdfv1 = np.searchsorted(data2, data1)/float(len(data2)) # this could be more efficient
+    cdfv2 = np.searchsorted(data1, data2)/float(len(data1)) # this could be more efficient
+    D = (np.amax(cdfv1-np.arange(len(data1))/float(len(data1))) + 
+            np.amax(cdfv2-np.arange(len(data2))/float(len(data2))))
+
+    Ne = len(data1)*len(data2)/float(len(data1)+len(data2))
+    return D, kuiper_FPP(D, Ne)
+
+
+
+def fold_intervals(intervals):
+    """Fold the weighted intervals to the interval (0,1).
+
+    Convert a list of intervals (ai, bi, wi) to a list of non-overlapping
+    intervals covering (0,1). Each output interval has a weight equal
+    to the sum of the wis of all the intervals that include it. All intervals
+    are interpreted modulo 1, and weights are accumulated counting 
+    multiplicity.
+
+    Parameters
+    ----------
+    intervals : list of three-element tuples (ai,bi,wi)
+        The intervals to fold; ai and bi are the limits of the interval, and
+        wi is the weight to apply to the interval.
+
+    Returns
+    -------
+    breaks : array of floats length N
+        The endpoints of a set of intervals covering [0,1]; breaks[0]=0 and
+        breaks[-1] = 1
+    weights : array of floats of length N-1
+        The ith element is the sum of number of times the interval 
+        breaks[i],breaks[i+1] is included in each interval times the weight
+        associated with that interval.
+
+    """
+    r = []
+    breaks = set()
+    tot = 0
+    for (a,b,wt) in intervals:
+        tot += (np.ceil(b)-np.floor(a))*wt
+        fa = a%1
+        breaks.add(fa)
+        r.append((0,fa,-wt))
+        fb = b%1
+        breaks.add(fb)
+        r.append((fb,1,-wt))
+
+    breaks.add(0.)
+    breaks.add(1.)
+    breaks = list(breaks)
+    breaks.sort()
+    breaks_map = dict([(f,i) for (i,f) in enumerate(breaks)])
+    totals = np.zeros(len(breaks)-1)
+    totals += tot
+    for (a,b,wt) in r:
+        totals[breaks_map[a]:breaks_map[b]]+=wt
+    return np.array(breaks), totals
+
+def cdf_from_intervals(breaks, totals):
```

Comment:
Would it not be easier to use `np.interp` instead of coding the algorithm yourself? Numpy routines are already well debugged and many of them are written in C to optimize for speed (I don't know if `np.interp` is one of them, but even if not that will probably be added in future numpy versions).